File: backend/app/models/shelf.py
from dataclasses import dataclass

from app.models.identifiers import OsmId


@dataclass
class Shelf:
    """Data class to represent shelf information."""

    osm_id: OsmId
    name: str | None
    latitude: float
    longitude: float
    address: str | None
    type: str | None
    operator: str | None
    website: str | None
    opening_hours: str | None
    osm_check_date: str | None
    osm_last_updated: str


@dataclass
class LocatedShelf:
    """Data class to represent a shelf with distance information."""

    shelf: Shelf
    distance_meters: float | None

File: frontend/src/types/Shelf.ts
import { removeOsmIdPrefix } from "../services/prefix";

export type Shelf = {
  osmId: string;
  name?: string;
  latitude: number;
  longitude: number;
  address?: string;
  type?: string;
  operator?: string;
  website?: string;
  openingHours?: string;
  osmCheckDate?: string; // When the shelf data was last verified on OpenStreetMap
  osmLastUpdated: string; // When any part of the shelf data was last updated or added on OpenStreetMap
};

export type LocatedShelf = {
  shelf: Shelf;
  distanceMeters: number | null;
};

export function getShelfRepresentation(
  shelf?: Shelf | null,
  fallback = "No shelf selected",
): string {
  return (
    shelf?.name ||
    shelf?.operator ||
    shelf?.address ||
    removeOsmIdPrefix(shelf?.osmId) ||
    fallback
  );
}
